### DISCUSSION TITLE
Changing the length scale of a mesh

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#16251] [Q&A Meshing] [answered]
Hi,

I am wondering if something is possible.

Say I have a mesh, whose solution has some length scale that is on the order of nanometers. So let's say the mesh is 3.0x3.0x3.0 nanometers with element size of 1 nm.

If I wanted to load this solution into a new calculation but rescale the lengths of the mesh to micrometers, 0.003x0.003x0.003 µm with an element size of 0.001 µm, is there an easy way of doing this?

John

**Comment by mangerij** [comment]
(besides rescaling my units in my equations by the way )

**Reply by YaqiWang** [reply]
How do you load the solution?

**Comment by mangerij** [comment]
Just as an initial condition: file = out.e in [Mesh] and then initial_from_file_var = var as an initial condition.

Would be great if there was something like rescale = 0.001 in [Mesh] to redefine the length coordinates in the out file but keeping solution variables. Thought i'd check if something like that existed
________________________________

**Reply by aeslaughter** [reply]
I think this can do what you need: https://mooseframework.inl.gov/source/meshgenerators/TransformGenerator.html

test/tests/meshgenerators/transform_generator/rotate_and_scale.i

**Reply by YaqiWang** [reply]
Maybe do this `initial_from_file_var` to a different variable and then add an IC to use that variable to initialize the target variable with a scaling factor in the input?

**Comment by aeslaughter** [comment]
Another thing to try is the SolutionUserObject, it has a scale option as well. This can be used with SolutionFunction for the IC I believe. It might take both the TransformGenerator and the SolutionUserObject. Good luck.

**Comment by mangerij** [comment, marked as answer]
TransfromGenerator looks to be it. Thanks! This might be very helpful in the near term.

**Comment by vincentlaboure** [comment]
I am guessing this approach would not be enough if the field you are trying to restart has units that contain length (e.g., a flux).
